### ISSUE TITLE
AOT compiling errors with 2.2.3

### ISSUE THREAD

**Issue by Tempus35** [swimlane/angular2-data-table#370]
**I'm submitting a ...**  (check one with "x")
```
[X ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here
```

**Current behavior**
When you try to compile your project in AOT mode it gives errors coming from Angular2-Data-Table.

ERROR in [default] /var/www/html/scheel_refactor/src/compiled/node_modules/angular2-data-table/release/components/body/body-row.component.ngfactory.ts:222:45 
Property '$colGroup' does not exist on type 'DataTableBodyRowComponent'.

ERROR in [default] /var/www/html/scheel_refactor/src/compiled/node_modules/angular2-data-table/release/components/body/body-row.component.ngfactory.ts:222:93 
Property '$colGroup' does not exist on type 'DataTableBodyRowComponent'.

ERROR in [default] /var/www/html/scheel_refactor/src/compiled/node_modules/angular2-data-table/release/components/header/header.component.ngfactory.ts:270:45 
Property 'colGroup' does not exist on type 'DataTableHeaderComponent'.

ERROR in [default] /var/www/html/scheel_refactor/src/compiled/node_modules/angular2-data-table/release/components/header/header.component.ngfactory.ts:270:92 
Property 'colGroup' does not exist on type 'DataTableHeaderComponent'.

ERROR in [default] /var/www/html/scheel_refactor/src/compiled/node_modules/angular2-data-table/release/directives/resizeable.directive.ngfactory.ts:71:34 
Supplied parameters do not match any signature of call target.

**Expected behavior**
The project compiles without errors from Angular2-Data-Table

**Reproduction of the problem**
Try compiling a project with Angular2-Data-Table as a lib in AOT mode

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**
Using https://github.com/qdouble/angular-webpack2-starter as the seed the lib was added to

* **Table version:** 0.7.x
2.2.3

* **Angular version:** 2.0.x
2.3.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
No clue

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by gitbufenshuo** [comment]
I got this issue too.
[typescript - 2.0.10]
[Angular - 2.3.0]
[Angular2-data-table - 2.2.3]
all beginning with the [Angular2 Hero Edit Seed].

**Comment by kuncevic** [comment]
I am having compilaton issues using **angular-cli: 1.0.0-beta.22-1**
That is what i've got when running `ng build`:
```
Cannot read property 'VisibilityDirective' of undefined, resolving symbol Angula
r2DataTableModule in c:/Projects/work/CorumCloud/Main/Source/Sapphire/Source/Sap
phire.UI/Source/node_modules/angular2-data-table/release/datatable.module.d.ts,
resolving symbol Angular2DataTableModule in c:/Projects/work/CorumCloud/Main/Sou
rce/Sapphire/Source/Sapphire.UI/Source/node_modules/angular2-data-table/release/
datatable.module.d.ts
```

**Comment by amcdnl** [comment]
Fixed in 2.3.0 ... release coming shortly.
